backend/app/models.py

```diff
@@ -413,3 +413,49 @@ def by_transaction(cls, db, txn: Transaction):
             .filter(StripeTransaction.transaction == txn.id)
             .first()
         )
+
+
+class UserOwnedApp(Base):
+    __tablename__ = "userownedapp"
+
+    app_id = Column(String, nullable=False, primary_key=True)
+    account = Column(
+        Integer,
+        ForeignKey(FlathubUser.id, ondelete="CASCADE"),
+        nullable=False,
+        primary_key=True,
+    )
+    created = Column(DateTime, nullable=False)
+
+    @staticmethod
+    def user_owns_app(db, user: FlathubUser, app_id: str):
+        return (
+            db.session.query(UserOwnedApp)
+            .filter_by(account=user.id, app_id=app_id)
+            .first()
+            is not None
+        )
+
+    @staticmethod
+    def all_by_user(db, user: FlathubUser):
```

Comment:
Maybe `all_owned_by_user` would be clearer?

Reply:
This will always be invoked with the class name on the front, so `UserOwnedApp.all_by_user(db, login["user"])` or similar. As such, I don't think repeating `owned` is necessary.